### ISSUE TITLE
Whitelabel Error Page - GreetingController annotation might be wrong

### ISSUE THREAD

**Issue by CarloAtDexels** [spring-guides/gs-serving-web-content#27]
When the `GreetingController` has the annotation `@Controller` annotation the URL to localhost:8080/greeting does not work. Changing the annotation to `@RestController` (as suggested in https://stackoverflow.com/questions/40797628/spring-boot-thymeleaf-viewresolver-on-rest-controllers) fixes the error. 

In the browser the error is shown as:

```
Whitelabel Error Page
This application has no explicit mapping for /error, so you are seeing this as a fallback.

Sun Dec 30 00:07:06 CET 2018
There was an unexpected error (type=Internal Server Error, status=500).
```

In the terminal the error:

```
2018-12-30 00:07:06.767 ERROR 4852 --- [nio-8080-exec-1] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is org.thymeleaf.exceptions.TemplateInputException: Error resolving template "greeting", template might not exist or might not be accessible by any of the configured Template Resolvers] with root cause

org.thymeleaf.exceptions.TemplateInputException: Error resolving template "greeting", template might not exist or might not be accessible by any of the configured Template Resolvers
```

**Comment by CarloAtDexels** [comment]
I used the complete sample, added the maven pom and then run Application. But if it occurs only for me I'll close this issue.

**Comment by dsyer** [comment]
When you say “added the maven pom” what does that mean exactly? There is already a pom.xml in the complete sample. If you use it, I expect it will work.

**Comment by CarloAtDexels** [comment]
I got it to work and figured out what went wrong!

I used the IntelliJ functionality VCS -> Checkout from version control -> Git to add an existing project. This does not add the pom.xml. Also the folder structure is different from the complete sample. It sets the main folder as the project root. Thymeleaf tries to search for the template under src/main/resources/templates when the template is under main/resources/templates. This caused the org.thymeleaf.exceptions.TemplateInputException
